### ISSUE TITLE
Confusing suggested configuration change

### ISSUE THREAD

**Issue by SoMuchForSubtlety** [fwupd/fwupd#6585]
**Describe the bug**
`fwupdmgr security` shows the following message
```
╔══════════════════════════════════════════════════════════════════════════════╗
║ Configuration Change Suggested: Encrypted RAM                                ║
╠══════════════════════════════════════════════════════════════════════════════╣
║ Encrypted RAM makes it impossible for information that is stored in device   ║
║ memory to be read if the memory chip is removed and accessed.                ║
║                                                                              ║
║ This tool can change the BIOS setting 'com.thinklmi.TSME' from 'Enable' to   ║
║ 'Enable' automatically, but it will only be active after restarting the      ║
║ computer.                                                                    ║
║                                                                              ║
║ You should ensure you are comfortable restoring the setting from the system  ║
║ firmware setup, as this change may cause the system to not boot into Linux   ║
║ or cause other system instability.                                           ║
╚══════════════════════════════════════════════════════════════════════════════╝
```

`from 'Enable' to  'Enable'` does not make sense. The output also reports `✘ Encrypted RAM: Not supported`, maybe that's related?

**Steps to Reproduce**
run `fwupdmgr security`

**fwupd version information**

```shell
compile   com.hughsie.libxmlb           0.3.14
compile   com.hughsie.libjcat           0.1.14
compile   org.freedesktop.fwupd         1.9.10
runtime   org.freedesktop.fwupd-efi     1.4
compile   org.freedesktop.gusb          0.4.8
runtime   com.hughsie.libjcat           0.1.14
runtime   org.freedesktop.gusb          0.4.8
runtime   org.freedesktop.fwupd         1.9.10
runtime   org.kernel                    6.6.8-200.fc39.x86_64
```

installed via `dnf`

<details>

<summary><b>fwupd device information</b></summary>

Please provide the output of the fwupd devices recognized in your system.

```
LENOVO 20UD0013GE
│
├─AMD Ryzen 7 PRO 4750U with Radeon Graphics:
│ │   Device ID:          4bde70ba4e39b28f9eab1628f9dd6e6244c03027
│ │   Current version:    0x08600109
│ │   Vendor:             Advanced Micro Devices, Inc.
│ │   GUIDs:              1f2e0c2f-e078-5fcb-a8de-f78eff9a470a ← CPUID\PRO_0&FAM_17&MOD_60
│ │                       5fd5eb41-dd4a-5742-8448-861eaaa61cf8 ← CPUID\PRO_0&FAM_17&MOD_60&STP_1
│ │   Device Flags:       • Internal device
│ │   Device Requests:    • Message
│ │ 
│ ├─Graphics Processing Unit (GPU):
│ │ │   Device ID:        e32966e41f54b47823eef28bd8b4bd9e1018b923
│ │ │   Summary:          Renoir Generic VBIOS
│ │ │   Current version:  017.010.000.031.000000
│ │ │   Vendor:           Advanced Micro Devices, Inc. [AMD/ATI] (PCI:0x1002)
│ │ │   GUID:             ffd1eae2-b5dd-58c4-b217-5ff61b0543b8 ← AMD\113-RENOIR-037
│ │ │   Device Flags:     • Internal device
│ │ │   Device Requests:  • Message
│ │ │ 
│ │ └─LP140WF9-SPF1:
│ │       Device ID:      aec1a869eb0df71b7cea6b3ac71d39b830faf164
│ │       GUID:           269df938-248e-5e93-8de5-c7046e71fa11 ← DRM\VEN_LGD&DEV_0608
│ │       Device Flags:   • Internal device
│ │       Device Requests:• Message
│ │     
│ ├─Secure Processor:
│ │     Device ID:        c54ab0237d7a8db8c717b68e0be78e4374a2a079
│ │     Current version:  00.11.00.81
│ │     Bootloader Version:00.11.00.81
│ │     Vendor:           Advanced Micro Devices, Inc. (PCI:0x1022)
│ │     GUIDs:            0e8dc554-a0a2-51fb-b439-1eb72b14ec38 ← PCI\VEN_1022&DEV_15DF
│ │                       66c84470-1ef2-55dd-89c2-3a240a2380b4 ← PCI\VEN_1022&DEV_15DF&SUBSYS_17AA5081
│ │     Device Flags:     • Internal device
│ │     Device Requests:  • Message
│ │   
│ └─System Management Unit (SMU):
│       Device ID:        db0330716216c629bb2c07256e5d018f499eb6ce
│       Summary:          Microcontroller used within CPU/APU program 0
│       Current version:  55.93.0
│       Vendor:           Advanced Micro Devices, Inc.
│       GUID:             165feb35-d368-5388-b2ab-c513021bf019 ← /sys/devices/platform/AMDI0005:00
│       Device Flags:     • Internal device
│       Device Requests:  • Message
│     
├─GPIO controller:
│     Device ID:          f685512aa07369c9e77742acef941d779d31e766
│     GUID:               37b440a9-2473-5087-a39b-db84f32a8ed8 ← GPIO\ID_AMDI0030:00
│   
├─Integrated Camera:
│     Device ID:          54f0d9041b6c5438c7ff825f5139559c5ca1b222
│     Current version:    58.18
│     Vendor:             Chicony Electronics Co.,Ltd. (USB:0x04F2)
│     Serial Number:      0001
│     GUID:               f8d483f9-9284-51b6-9363-7e6d59c40a97 ← USB\VID_04F2&PID_B6D0
│     Device Flags:       • Updatable
│     Device Requests:    • Image
│   
├─MZVLB512HBJQ-000L7:
│     Device ID:          71b677ca0f1bc2c5b804fa1d59e52064ce589293
│     Summary:            NVM Express solid state drive
│     Current version:    5M2QEXF7
│     Vendor:             Samsung (NVME:0x144D)
│     Serial Number:      S4ENNX1N609178
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUIDs:              47335265-a509-51f7-841e-1c94911af66b ← NVME\VEN_144D&DEV_A808
│                         c9d531ea-ee7d-5562-8def-c64d0d144813 ← NVME\VEN_144D&DEV_A808&SUBSYS_144DA801
│                         f4ceb2a8-09c4-55c0-af2a-13ccd63925b1 ← SAMSUNG MZVLB512HBJQ-000L7
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│                         • Signed Payload
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─Prometheus:
│ │   Device ID:          50efef912094e784a03628350441f6927c4920b0
│ │   Summary:            Fingerprint reader
│ │   Current version:    10.01.3478575
│ │   Vendor:             Synaptics (USB:0x06CB)
│ │   Install Duration:   2 seconds
│ │   Serial Number:      219223778239572
│ │   GUID:               8088f861-6318-5b1e-9ce4-fbddbedb09ac ← USB\VID_06CB&PID_00BD
│ │   Device Flags:       • Updatable
│ │                       • Supported on remote server
│ │                       • Cryptographic hash verification is available
│ │                       • Signed Payload
│ │   Device Requests:    • Image
│ │ 
│ └─Prometheus IOTA Config:
│       Device ID:        30469829b6adc759baf3b0c3cf19454985aa73a2
│       Summary:          Fingerprint reader config
│       Current version:  0022
│       Minimum Version:  0022
│       Vendor:           Synaptics (USB:0x06CB)
│       GUIDs:            7c5a1e62-38fa-5859-9337-09dbac6377e4 ← USB\VID_06CB&PID_00BD-cfg
│                         21a62213-7012-552c-9f1a-a51d07c87242 ← USB\VID_06CB&PID_00BD&CFG1_3462&CFG2_0
│       Device Flags:     • Updatable
│                         • Supported on remote server
│                         • Only version upgrades are allowed
│                         • Signed Payload
│       Device Requests:  • Image
│     
├─System Firmware:
│ │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
│ │   Summary:            UEFI ESRT device
│ │   Current version:    0.1.46
│ │   Vendor:             Lenovo (DMI:LENOVO)
│ │   Update State:       Success
│ │   Problems:           [31m[1m• Device requires AC power to be connected[0m
│ │   GUID:               01453b71-da0c-4832-9f4f-e378245339c7
│ │   Device Flags:       • Internal device
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Needs a reboot after installation
│ │                       • Cryptographic hash verification is available
│ │                       • Device is usable for the duration of the update
│ │                       • Updatable
│ │   Device Requests:    • Message
│ │                       • Image (custom)
│ │ 
│ └─UEFI dbx:
│       Device ID:        362301da643102b9f38477387e2193e57abaa590
│       Summary:          UEFI revocation database
│       Current version:  371
│       Minimum Version:  371
│       Vendor:           UEFI:Linux Foundation
│       Install Duration: 1 second
│       GUIDs:            5971a208-da00-5fce-b5f5-1234342f9cf7 ← UEFI\CRT_A9087D1044AD18F7A94916D284CBC01827CF23CD8F60B79072C9CAA1FEF4D649&ARCH_X64
│                         f8ba2887-9411-5c36-9cee-88995bb39731 ← UEFI\CRT_A1117F516A32CEFCBA3F2D1ACE10A87972FD6BBE8FE0D0B996E09E65D802A503&ARCH_X64
│       Device Flags:     • Internal device
│                         • Updatable
│                         • Supported on remote server
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Only version upgrades are allowed
│                         • Signed Payload
│       Device Requests:  • Message
│                         • Image
│     
├─TPM:
│     Device ID:          c6a80ac3a22083423992a3cb15018989f37834d6
│     Current version:    1.258.0.0
│     Vendor:             ST Microelectronics (TPM:STM)
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUIDs:              84df3581-f896-54d2-bd1a-372602f04c32 ← TPM\VEN_STM&DEV_0001
│                         bfaed10a-bbc1-525b-a329-35da2f63e918 ← TPM\VEN_STM&MOD_
│                         70b7b833-7e1a-550a-a291-b94a12d0f319 ← TPM\VEN_STM&DEV_0001&VER_2.0
│                         06f005e9-cb62-5d1a-82d9-13c534c53c48 ← TPM\VEN_STM&MOD_&VER_2.0
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device can recover flash failures
│                         • Full disk encryption secrets may be invalidated when updating
│                         • Signed Payload
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     Device ID:          349bb341230b1a86e5effe7dfe4337e1590227bd
│     Summary:            UEFI ESRT device
│     Current version:    1057227777
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               4b4c862d-abcd-4a8d-a386-15644b404a7b
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     Device ID:          2292ae5236790b47884e37cf162dcf23bfcd1c60
│     Summary:            UEFI ESRT device
│     Current version:    17011397
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               cc484fef-8758-41ce-b42f-f402f79881ac
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     Device ID:          f95c9218acd12697af946874bfe4239587209232
│     Summary:            UEFI ESRT device
│     Current version:    65582
│     Minimum Version:    65582
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               4bea12df-56e3-4cdb-97dd-f133768c9051
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     Device ID:          d96de5c124b60ed6241ebcb6bb2c839cb5580786
│     Summary:            UEFI ESRT device
│     Current version:    65794
│     Minimum Version:    65794
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               42911c88-933b-43f5-b737-c91c20779af7
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│     Device Requests:    • Message
│                         • Image (custom)
│   
├─UEFI Device Firmware:
│     Device ID:          f37fb01122dd62c773f4e84ec89737e059712d59
│     Summary:            UEFI ESRT device
│     Current version:    22552
│     Minimum Version:    1
│     Vendor:             DMI:LENOVO
│     Update State:       Success
│     Problems:           [31m[1m• Device requires AC power to be connected[0m
│     GUID:               58ac1529-d706-459c-9f98-c711bb777410
│     Device Flags:       • Internal device
│                         • System requires external power source
│                         • Needs a reboot after installation
│                         • Device is usable for the duration of the update
│                         • Updatable
│     Device Requests:    • Message
│                         • Image (custom)
│   
└─UEFI Platform Key:
      Device ID:          6924110cde4fa051bfdc600a60620dc7aa9d3c6a
      Summary:            Lenovo Ltd. PK CA 2012
      Vendor:             Lenovo Ltd.
      GUID:               71599d14-9b31-5270-b3bd-74c494585820 ← UEFI\CRT_9AEF2123F4DE7C19AFABD909BB2C8CAC4411E07E
```

</details>

**Additional questions**

- Operating system and version: Fedora 39
- Have you tried rebooting? yes
- Is this a regression? ¯\\\_(ツ)\_/¯

**Comment by superm1** [comment]
Can you please provide output from

`sudo fwupdtool security -vv` to better understand this issue?

**Comment by SoMuchForSubtlety** [comment]
[security vv.log](https://github.com/fwupd/fwupd/files/13802233/security.vv.log)

**Comment by superm1** [comment]
A few things:
1. Can you check what `cat /sys/bus/pci/drivers/ccp/*/tsme_status` returns?  This will help clarify if the problem is the obsoleting of MSR and PCI_PSP plugins on your system.
2 Can you please share `/proc/cmdline`?
3. Can you please try https://github.com/fwupd/fwupd/pull/6587?  I think this will stop it offering to change the setting at least.

**Comment by SoMuchForSubtlety** [comment]
> 1. Can you check what `cat /sys/bus/pci/drivers/ccp/*/tsme_status` returns? 

```
0
```

> 2. Can you please share `/proc/cmdline`?

```
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-6.6.8-200.fc39.x86_64 root=UUID=497dde77-e636-410c-a2f6-3e7cda1a6dca ro rootflags=subvol=root rhgb quiet rhgb quiet amd_pstate=passive preempt=full
```

>  3. Can you please try #6587

I followed the instructions from the wiki, but get the following error when running the newly installed version
```
fwupdmgr: error while loading shared libraries: libfwupd.so.3: cannot open shared object file: No such file or directory
```

**Comment by superm1** [comment]
Fedora doesn't set /usr/local/lib64 by default in the library search path.

Try running with 

`LD_LIBRARY_PATH=/usr/local/lib64` in your environment

**Comment by SoMuchForSubtlety** [comment]
Thanks, I can confirm that the change suggestion no longer appears.

**Comment by superm1** [comment]
Okay. Then I'll backport that to 1_9_X.

Regarding the reason it suggested it though in the first place; that's pretty odd.

Can you please reset BIOS default settings and then try to turn it back on again?

**Comment by SoMuchForSubtlety** [comment]
After resetting the BIOS, fwupd reported `✘ Encrypted RAM: Not supported` again, but still offered to enable it. I applied the setting and after a reboot, fwupd now reports `✔ Encrypted RAM: Encrypted`.

**Comment by superm1** [comment]
Okay then there is definitely some bug in the BIOS. I'll fix the fwupd side of it, but you should report the BIOS side of it to Lenovo.

**Comment by superm1** [comment]
Fixed fwupd side of the issue in d74375a19c4edfa35792bb88159a754351841741 on 1_9_X and e8c45d7e52945917601e9260bc5e5986bab6abb9 in main.
